test/functional/test-xhr.js

```diff
@@ -73,7 +75,17 @@ describe.configure().skipSafari().run('XHR', function() {
   beforeEach(() => {
     sandbox = sinon.sandbox.create();
     ampdocServiceForStub = sandbox.stub(Services, 'ampdocServiceFor');
-    ampdocServiceForStub.returns({isSingleDoc: () => false});
+    ampdocViewerStub = sandbox.stub(Services, 'viewerForDoc');
+    ampdocViewerStub.returns({
+      whenFirstVisible: () => {
+        return Promise.resolve();
+      },
+    });
+    ampdocServiceForStub.returns({
+      isSingleDoc: () => false,
+      getAmpDoc: () => { return ampdocViewerStub; },
```

Comment:
Not a huge deal since this is a test file however I believe you could reduce this down a bit.
```
ampdocServiceForStub.returns({
  isSingleDoc: () => false,
  getAmpDoc: () => ampdocViewerStub,
});